app/src/main/java/com/example/memeitupapp/ui/gridmemes/GridMemesFragment.kt

```diff
@@ -6,13 +6,35 @@ import android.view.View
 import android.view.ViewGroup
 import androidx.fragment.app.Fragment
 import com.example.memeitupapp.R
+import com.example.memeitupapp.data.entity.Meme
+import com.example.memeitupapp.util.DEFAULT_INT
+import com.example.memeitupapp.util.Data
+import com.example.memeitupapp.util.Event
+import kotlinx.android.synthetic.main.fragment_grid_memes.activity_main_call_service_button
+import kotlinx.android.synthetic.main.fragment_grid_memes.activity_main_first_meme_name
 
 class GridMemesFragment : Fragment() {
 
+    private lateinit var gridMemesViewModel: GridMemesViewModel
+
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
         inflater.inflate(R.layout.fragment_grid_memes, container, false)
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        gridMemesViewModel = GridMemesViewModel()
```

Comment:
this is not the way of intantiate viewmodel. check it please.
i bet your data is lost when configuration changes

Reply:
fixed, thanks for the help